scripts/validate_data_integrity.py

```
@@ -2,22 +2,158 @@
+                   COUNT(CASE WHEN prev_hash IS NULL AND sequence_number > 1 THEN 1 END) as broken_links
+            FROM fsma.cte_events
+            WHERE prev_hash IS NOT NULL OR sequence_number = 1
+        """))
+        row = result.fetchone()
+        if row is None:
+            return True, "No CTE events found (empty chain — OK)"
+        total, broken = row[0], row[1]
+        if broken > 0:
+            return False, f"Hash chain broken: {broken}/{total} events have missing prev_hash"
+        return True, f"Hash chain intact: {total} events verified"
+    except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)